examples/tutorials/PytorchLightning_Integration.ipynb

```diff
@@ -0,0 +1,464 @@
+    "\n",
+    "1. Multi-gpu training functionalities: pytorch-lightning provides easy multi-gpu, multi-node training. It also simplifies the process of launching multi-gpu, multi-node jobs across different cluster infrastructure, e.g. AWS, slurm based clusters.\n",
+    "\n",
+    "1. Reducing boilerplate pytorch code: lightning takes care of details like, `optimizer.zero_grad(), model.train(), model.eval()`. Lightning also provides experiment logging functionality, for e.g. irrespective of training on CPU, GPU, multi-nodes the user can use the method `self.log` inside the trainer and it will appropriately log the metrics.\n",
+    "1. Features that can speed up training: half-precision training, gradient checkpointing, code profiling.\n",
+    "\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1VVLqq0vMlPkSEXeqcFnHY_zEuvDOQu50?usp=sharing)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bbb99230",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "- This notebook assumes that you have already installed deepchem, if you have not follow the instructions at the deepchem installation page: https://deepchem.readthedocs.io/en/latest/get_started/installation.html.\n",
+    "- Install pytorch lightning following the instructions on lightning's home page: https://www.pytorchlightning.ai/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "98ee91c0",
+   "metadata": {},
+   "source": [
+    "Import the relevant packages."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "6890f523",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import deepchem as dc\n",
+    "from deepchem.models import GCNModel\n",
+    "import pytorch_lightning as pl\n",
+    "import torch\n",
+    "from torch.nn import functional as F\n",
+    "from torch import nn\n",
+    "import pytorch_lightning as pl\n",
+    "from pytorch_lightning.core.lightning import LightningModule\n",
+    "from torch.optim import Adam\n",
+    "import numpy as np\n",
+    "import torch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe62d7c3",
+   "metadata": {},
+   "source": [
+    "## Deepchem Example\n",
+    "\n",
+    "Below we show an example of a Graph Convolution Network (GCN). Note that this is a simple example which uses a GCNModel to predict the label from an input sequence. We do not showcase the complete functionality of deepchem in this example as we want to restructure the deepchem code and adapt it so that it can be easily plugged into pytorch-lightning. This example was inspired from the `GCNModel` documentation present [here](https://github.com/deepchem/deepchem/blob/a68f8c072b80a1bce5671250aef60f9cc8519bec/deepchem/models/torch_models/gcn.py#L200)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2859f97b",
+   "metadata": {},
+   "source": [
+    "**Prepare the dataset**"
```

Comment:
For this and the following sections, we should have a sentence description of what we are doing for first time readers or new DeepChem users.